kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/delayed/DelayedOperationPurgatory.java

```diff
@@ -340,7 +340,7 @@ public void watch(T t) {
         }
 
         // traverse the list and try to complete some watched elements
-        public int tryCompleteWatched() {
+        public int tryCompleteWatched(Object key) {
```

Comment:
Could you explain why did you add an argument here? Before this PR, the `key` is a field of `Watchers` and is used in following code

```java
            if (operations.isEmpty()) {
                removeKeyIfEmpty(key, this);
            }
```

However, after this change, the `key` no longer represents the field of `Watchers`. Instead, it's passed by `DelayedOperationPurgatory#checkAndComplete`. Then the behavior changes. Could you explain the difference of these two `key`s?

Reply:
I added this in order to have a way in the future to pass to wakeup the motivation of the trigger.
this patch currently triggers a new Fetch execution for every partition that is involved, but in the future we can make it better and re-read the data only from the partition that received a message.
I didn't want to complicate this patch.
I can remove this parameter here and I will add it back in a follow up patch (that I still haven't implemented, because it sounds premature to me, at least for my tests and use cases)